src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cssProperties = {};
+  const arrayOfItems = sourceString
+    .split(';') // ['ключ : свойство','ключ:свойство']
+    .filter((item) => item.includes(':')) // убираем  пустые элементы
+    .map((elem) => elem.split(':')); //[[ключ, свойство], [ключ, свойство]]
+
+  arrayOfItems.forEach((unit) => {
+    const key = unit[0].trim();
+    const value = unit[1].trim();
+
+    cssProperties[key] = value;
+  });
+  return cssProperties;
```

Comment:
The use of `forEach` to iterate and build the `cssProperties` object is fine, as it doesn't return a new array and is an exception to the rule mentioned in the [CODE KNOWLEDGE] checklist.